### PR TITLE
Source Zendesk Talk: enable `high` test strictness level in SAT

### LINKED ISSUE #19072: Source Zendesk Talk: enable `high` test strictness level in SAT
## What
A `test_strictness_level` field was introduced to Source Acceptance Tests (SAT).
Zendesk Talk is a generally_available connector, we want it to have a `high` test strictness level.

**This will help**:
- maximize the SAT coverage on this connector.
- document its potential weaknesses in term of test coverage.

## How
1. Migrate the existing `acceptance-test-config.yml` file to the latest configuration format. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/README.md#L61))
2. Enable `high` test strictness level in `acceptance-test-config.yml`. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/testing-connectors/source-acceptance-tests-reference.md#L240))
3. Commit changes on `acceptance-test-config.yml` and open a PR.
4. Run SAT with the `/test` command on the branch.
5. If tests are failing please fix the failing test or use `bypass_reason` fields to explain why a specific test can't be run.

### PR DESCRIPTION
**Pull request by lazebnyi**
https://github.com/airbytehq/airbyte/issues/19072
SAT high strictness

### CODE REVIEW

File: airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
...lls", "data": {"id": 360121166995, "created_at": "2021-10-20T13:22:25Z", "updated_at": "2021-10-20T16:22:34Z", "agent_id": null, "call_charge": "0.003", "consultation_time": 0, "completion_status": "completed", "customer_id": null, "customer_requested_voicemail": false, "direction": "outbound", "duration": 6, "exceeded_queue_wait_time": false, "hold_time": 0, "minutes_billed": 1, "outside_business_hours": false, "phone_number_id": 360000121575, "phone_number": "+12059531462", "ticket_id": null, "time_to_answer": null, "voicemail": false, "wait_time": 0, "wrap_up_time": 0, "ivr_time_spent": null, "ivr_hops": null, "ivr_destination_group_name": null, "talk_time": 0, "ivr_routed_to": null, "callback": false, "callback_source": null, "default_group": false, "ivr_action": null, "overflowed": false, "overflowed_to": null, "recording_control_interactions": 0, "recording_time": 0, "not_recording_time": 0, "call_recording_consent": "always", "call_recording_consent_action": null, "call_recording_consent_keypress": null, "call_group_id": null, "call_channel": null, "quality_issues": ["none"], "line": "+12059531462", "line_id": 360000121575, "line_type": "phone"}, "emitted_at": 1674159472102}
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl:10]: Can you confirm that 12059531462 is our number, and not the customer number?

File: airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
..., "wait_en", "hold_en", "callback_en", "callback-confirmation_en", "call-recording-opt-out_en", "call-recording-opt-in_en"], "categorised_greetings": {"1": "voicemail_en", "2": "available_en", "3": "wait_en", "4": "hold_en", "6": "callback_en", "7": "callback-confirmation_en", "8": "call-recording-opt-out_en", "9": "call-recording-opt-in_en"}, "categorised_greetings_with_sub_settings": {"1": {"voicemail_on_outside_business_hours": "voicemail_en", "voicemail_on_inside_business_hours": "voicemail_en", "voicemail_off_inside_business_hours": "voicemail_en_voicemail_config", "voicemail_off_outside_business_hours": "voicemail_en_voicemail_config"}, "2": {"voicemail_on": "available_en", "voicemail_off": "available_en_voicemail_config"}, "3": "wait_en", "4": "hold_en", "6": "callback_en", "7": "callback-confirmation_en", "8": "call-recording-opt-out_en", "9": "call-recording-opt-in_en"}, "sms_group_id": null, "capabilities": {"sms": true, "mms": true, "voice": true, "emergency_address": true}, "sms_enabled": false, "voice_enabled": true, "priority": 0, "outbound_enabled": true, "line_type": "phone", "ivr_id": null, "schedule_id": null, "failover_number": null}, "emitted_at": 1674159481130}
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl:81]: Yep - this seems to be the account's number +1 (205) 953-1462

**Review by evantahler — APPROVED** [review]
I think this looks good!

File: airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl (hunk 1 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl:10]: It look like account number. 
@ycherniaiev @igrankova can you help us with that?

File: airbyte-integrations/connectors/source-zendesk-talk/acceptance-test-config.yml [hunk 3]
```diff
@@ -2,32 +2,52 @@
...
+          extra_fields: no
+          exact_order: no
+          extra_records: yes
+        empty_streams:
+          - name: "account_overview"
+            bypass_reason: "The objects can mutate quite often."
+          - name: "agents_overview"
+            bypass_reason: "The objects can mutate quite often."
+          - name: "current_queue_activity"
+            bypass_reason: "The objects can mutate quite often."
```
[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-zendesk-talk/acceptance-test-config.yml:40]: This streams has `current_timestamp` and it updates every time when we get record.

[the remark this answers is not in the document]
Comment (evantahler) [reply airbyte-integrations/connectors/source-zendesk-talk/acceptance-test-config.yml:40]: Can we ignore just the problem field? https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-faker/acceptance-test-config.yml#L29-L34

[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-zendesk-talk/acceptance-test-config.yml:40]: And other streams has a lot of data, so we need `extra_records` and `extra_fields` at the same time. But it is impossible.

[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-zendesk-talk/acceptance-test-config.yml:40]: I'm not sure that we have `ignored_fields` for basic read test

**Review by darynaishchenko — APPROVED** [review]

File: airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl (hunk 1 above)
Comment (ycherniaiev) [reply airbyte-integrations/connectors/source-zendesk-talk/integration_tests/expected_records.jsonl:10]: I checked this number everywhere through docs and public resources. I have no clues to think that it is a real number

**Pull request merged**